extensions/amp-analytics/0.1/events.js

```diff
@@ -667,9 +682,36 @@ class TimerEventHandler {
     }
     win.clearInterval(this.intervalId_);
     this.intervalId_ = undefined;
+    this.calculateDuration_();
+    this.lastPingTime_ = undefined;
+    this.intervalCallback_();
+    this.intervalCallback_ = null;
     this.unlistenForStop_();
     this.listenForStart_();
   }
+
+  /** @private */
+  calculateDuration_() {
+    if (this.startTime_) {
+      this.timerDuration_ +=
+          Date.now() - (this.lastPingTime_ || this.startTime_);
+    }
+  }
+
+  /** @return {{timerDuration: number, timerStart: number}} */
+  reportTimerVars() {
+    if (this.isRunning()) {
+      this.calculateDuration_();
+      this.lastPingTime_ = Date.now();
+    }
+    const durationSeconds = Math.floor(this.timerDuration_ / 1000);
+    // Keep track of partial unreported seconds so they can roll over later.
+    this.timerDuration_ -= (durationSeconds * 1000);
```

Comment:
is the state introduced only to capture the fraction? can we report in `ms`?

Reply:
Done. We can move this into the extension.